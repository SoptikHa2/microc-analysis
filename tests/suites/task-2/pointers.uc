f(x) {
   return &x;
 }

 g(y) {
   return *y;
 }
