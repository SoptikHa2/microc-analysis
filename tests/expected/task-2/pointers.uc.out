---
test: type suites/task-2/pointers.uc
---
⟦f(x[1:3]){...}:[1:1]⟧ = (t1) -> ↑t1
⟦x[1:3]⟧ = t1
⟦g(y[5:4]){...}:[5:2]⟧ = (↑t2) -> t2
⟦y[5:4]⟧ = ↑t2

exit: 0
